### PR TITLE
refactor PnL

### PR DESCRIPTION
**Pull request by tarcisiozf**
* calculate pnl per trade, position, and strategy
* split realized and unrealized pnl
* rename fields

### CODE REVIEW

**Review by avsek477 — CHANGES_REQUESTED** [review]
other than fee changes, rest lgtm

File: lib/pnl/calc_unrealized_position_pnl.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+'use strict'
+
+const BigNumber = require('bignumber.js')
+
+module.exports = (position, currentPrice) => {
+  let sum = new BigNumber(0)
+  let volume = new BigNumber(0)
+
+  for (const { amount, price } of position.inventory) {
+    sum = sum.plus(price.multipliedBy(volume))
```
[the remark this answers is not in the document]
Comment (tarcisiozf) [reply lib/pnl/calc_unrealized_position_pnl.js:10]: good catch

File: lib/pnl/calc_realized_trade_pnl.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  let remainingAmount = amount.abs()
+
+  while (!remainingAmount.isZero()) {
+    const prevBuy = position.inventory.shift()
+    const priceDiff = price.minus(prevBuy.price)
+
+    if (prevBuy.amount.lessThan(remainingAmount)) {
+      remainingAmount = remainingAmount.minus(prevBuy.amount)
+      pnl = pnl.plus(priceDiff.multipliedBy(prevBuy.amount))
```
[the remark this answers is not in the document]
Comment (tarcisiozf) [reply lib/pnl/calc_realized_trade_pnl.js]: 7e16da1b72c613ef7ffffab7cd1aa48382660377

**Review by avsek477 — APPROVED** [review]
Lgtm

**Pull request merged**